protocol/x/clob/e2e/liquidation_deleveraging_test.go

```diff
@@ -67,6 +78,134 @@ func TestPlacePerpetualLiquidation_Deleveraging(t *testing.T) {
+			},
+			liquidatableSubaccountIds: []satypes.SubaccountId{constants.Carl_Num0},
+			liquidationConfig:         constants.LiquidationsConfig_FillablePrice_Max_Smmr,
+			clobPairs:                 []clobtypes.ClobPair{constants.ClobPair_Btc},
+
+			expectedSubaccounts: []satypes.Subaccount{
+				{
+					Id: &constants.Carl_Num0,
+					AssetPositions: []*satypes.AssetPosition{
+						{
+							AssetId: 0,
+							// Asset position + fill price - liquidation insurance fund fee
+							Quantums: dtypes.NewInt(-40_001_000_000 + 50_000_000_000 - 250_000_000),
+						},
+					},
+				},
+				{
+					Id: &constants.Dave_Num0,
+					PerpetualPositions: []*satypes.PerpetualPosition{
+						{
+							PerpetualId:  0,
+							Quantums:     dtypes.NewInt(200_000_000),
+							FundingIndex: dtypes.NewInt(0),
+						},
+					},
+				},
+			},
+		},
+		`Skip liquidating subaccount when TNC equals MMR with non-linear margin requirement`: {
+			liquidityTiers: []perptypes.LiquidityTier{
+				{
+					Id:                     3,
+					Name:                   "3",
+					InitialMarginPpm:       200_000,        // 20% IMF
+					MaintenanceFractionPpm: 500_000,        // 10% MMF
+					BasePositionNotional:   12_500_000_000, // $12.5k
+					ImpactNotional:         2_500_000_000,
+				},
+			},
+			perpetuals: []perptypes.Perpetual{
+				// Uses liquidity tier id 3
+				constants.BtcUsd_20PercentInitial_10PercentMaintenance,
+			},
+			subaccounts: []satypes.Subaccount{
+				{ // TNC = $10k, MMR = $10k
+					Id: &constants.Carl_Num0,
+					AssetPositions: []*satypes.AssetPosition{
+						{
+							AssetId:  0,
+							Quantums: dtypes.NewInt(-40_000_000_000), // $-40,000
+						},
+					},
+					PerpetualPositions: []*satypes.PerpetualPosition{
+						{
+							PerpetualId: 0,
+							Quantums:    dtypes.NewInt(100_000_000), // 1 BTC
+						},
+					},
+				},
+				constants.Dave_Num0_1BTC_Long_50000USD,
+			},
+			placedMatchableOrders: []clobtypes.MatchableOrder{
+				&constants.LongTermOrder_Dave_Num0_Id0_Clob0_Buy1BTC_Price50000_GTBT10,
+			},
+			liquidatableSubaccountIds: []satypes.SubaccountId{constants.Carl_Num0},
+			liquidationConfig:         constants.LiquidationsConfig_FillablePrice_Max_Smmr,
+			clobPairs:                 []clobtypes.ClobPair{constants.ClobPair_Btc},
+
+			// Provided liquidatable subaccount should be skipped in PrepareCheckState for not being liquidatable.
+			// This means no liquidation order placement / match will occur.
+			expectedSubaccounts: []satypes.Subaccount{
+				{
+					Id: &constants.Carl_Num0,
+					AssetPositions: []*satypes.AssetPosition{
+						{
+							AssetId:  0,
+							Quantums: dtypes.NewInt(-40_000_000_000), // $-40,000
+						},
+					},
+					PerpetualPositions: []*satypes.PerpetualPosition{
+						{
+							PerpetualId:  0,
+							Quantums:     dtypes.NewInt(100_000_000), // 1 BTC
+							FundingIndex: dtypes.NewInt(0),
+						},
+					},
+				},
```

Comment:
is this going to be used for other tests in the future? should we move it to testutil/constants?

Reply:
idk it's pretty specific to this test, don't really see it as a generally re-usable subaccount because it assumes non-linear margin requirements for its Asset and Perpetual positions to be relevant and frequently used values in tests